frontend/plan/src/reducers/schedule.js

```diff
@@ -89,6 +89,57 @@ const nextAvailable = (scheduleName, used) => {
 export const schedule = (state = initialState, action) => {
     const { cartSections } = state;
     switch (action.type) {
+        case SET_SCHEDULE_ID:
+            return {
+                ...state.schedules,
+                [action.scheduleName]: {
+                    ...state.schedules[action.scheduleName],
+                    id: action.id
+                }
+            };
+        case UPDATE_SCHEDULES:
+            // eslint-disable-next-line
+            const { schedulesFromBackend } = action;
+            // eslint-disable-next-line
+            const newScheduleObject = { ...state.schedules };
+            // eslint-disable-next-line
+            const newCart = [...state.cartSections];
+            if (schedulesFromBackend) {
+                schedulesFromBackend.forEach(({ id: scheduleId, title, sections, semester }) => {
+                    if (title === "cart") {
```

Comment:
Might want some documentation here about how the cart is stored as a schedule on the backend.